### PR TITLE
gRPC client config support

### PR DESCRIPTION
**Pull request by wtlucy**
Adds a new optional top-level server.xml element `<grpcTarget/>`, which can be used to set properties for gRPC services managed via `grpcClient-1.0`.  A single `grpcTarget` configuration can apply to one or many services.  Currently the following properties can be set:

* `target` = required string
    * outbound target service or RPC name used by app code
    * examples: `“/service.name/someMethod”`, `“/service.name/*”`, `"*"`
* `authnToken` = optional string
    * specify desired auth to apply: currently limited to `jwt` and `oauth`
* `headersToPropagate` = optional string
    * specify a list of header names that Liberty should attempt to propagate
* `enableKeepAlive` = optional boolean
* `keepAliveTime` = optional int
* `keepAliveTimeout` = optional int
* `maxInboundMessageSize` = optional int
* `clientInterceptors` = optional string {list of fully qualified class names}
    * specify interceptors that will be applied to this service
    * this list is merged with the annotation-defined list at runtime
* `sslConfig` = optional string
    * the ID of the SSL configuration to use
    * implementation is incomplete

For #10089 and #10090

### CODE REVIEW

**Comment by wtlucy** [comment]
#build

**Comment by wtlucy** [comment]
#build

**Comment by wtlucy** [comment]
#build

**Comment by wtlucy** [comment]
#build

File: dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+###############################################################################
+# -------------------------------------------------------------------------------------------------
+#CMVCPATHNAME com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties
+#ISMESSAGEFILE FALSE
+#NLS_ENCODING=UNICODE
+#NLS_MESSAGEFORMAT_NONE
+
+
+clientConfig=GRPC Client Properties
+clientConfig.desc=Configuration properties to be applied to gRPC targets that match the specified URI when they are constructed
```
Comment (dmuelle) [inline dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties:19]: Does "when they are constructed" refer to the time that the configuration properties are applied or the time that gRPC targets  match the specified URI?

File: dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+target.desc=The remote gRPC target URI, with wildcard support.
+
+headersToPropagate=HTTP header names to propagate
+headersToPropagate.desc=HTTP header names to propagate from an inbound request to an outbound call.
+
+clientInterceptors=GRPC client Interceptors
+clientInterceptors.desc=A list of fully qualified class names for gRPC client interceptor classes.
+
+authnToken=Authorization token
+authnToken.desc=The type of authorization token to propagate.  Currently this must be "jwt" or "oauth".
```
Comment (dmuelle) [inline dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties:31]: are quotes necessary?

Currently supported types are jwt or oauth

File: dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+enableKeepAlive=Keep alive
```
Comment (dmuelle) [inline dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties:33]: this is news to me but IBM terminology specifies keepalive, one word. Here's the guidance-

keepalive
adjective
Pertaining to the amount of time that elapses before a message is sent to verify a connection.

noun
A TCP/IP mechanism where a small packet is sent across the network at predefined intervals to determine whether the socket is still working correctly.

If this description applies to the function, update all instances to a single-word term

File: dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+keepAliveTime.desc=The time to wait for new messages before sending a new keep alive ping.
+
+keepAliveTimeout=Keep alive timeout
+keepAliveTimeout.desc=End the connection if a keepalive response is not received within this time.
+
+maxInboundMessageSize=Maximum message size
+maxInboundMessageSize.desc=The maximum inbound message size.
+ 
+sslConfig=The id of an ssl reference in server.xml
+sslConfig.desc=An id of an ssl reference found in server.xml.  That ssl configuration specified by that reference is used.
```
Comment (dmuelle) [inline dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties:46]: An id of an ssl reference found in server.xml. The ssl configuration that is specified by that reference is used.

Single space between these sentences.

File: dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+ 
+other=Any other variables
+other.desc=Any other variables can also be specified and will be passed to the GrpcTarget intact.
```
Comment (dmuelle) [inline dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties:49]: avoid future tense.

Any other variables that can  be specified and passed to the GrpcTarget intact.

File: dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties (hunk 1 above)
Comment (wtlucy) [reply dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties:19]: This isn't phrased well at all - I used `com.ibm.ws.jaxrs.2.x.config/resources/OSGI-INF/l10n/metatype.properties` as a guide.  

I'd intended for "when they are constructed" to apply to "the time that gRPC targets match the specified URI".  Would removing it completely be more clear?
`Configuration properties to be applied to gRPC targets that match the specified URI`

File: dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties (hunk 3 above)
Comment (wtlucy) [reply dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties:33]: I agree, this should be one word.  I'll update.

File: dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+target.desc=The remote gRPC target URI, with wildcard support.
+
+headersToPropagate=HTTP header names to propagate
+headersToPropagate.desc=HTTP header names to propagate from an inbound request to an outbound call.
+
+clientInterceptors=GRPC client Interceptors
+clientInterceptors.desc=A list of fully qualified class names for gRPC client interceptor classes.
+
+authnToken=Authorization token
+authnToken.desc=The type of authorization token to propagate. Currently this must be jwt or oauth.
```
Comment (dmuelle) [inline dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties:31]: Avoid "this" as the subject. Would the following be correct?

Currently supported token types are jwt or oauth

File: dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+keepAliveTime.desc=The time to wait for new messages before sending a new keepalive ping.
+
+keepAliveTimeout=Keepalive timeout
+keepAliveTimeout.desc=End the connection if a keepalive response is not received within this time.
+
+maxInboundMessageSize=Maximum message size
+maxInboundMessageSize.desc=The maximum inbound message size.
+ 
+sslConfig=The id of an ssl reference in server.xml
+sslConfig.desc=An id of an ssl reference found in server.xml. The ssl configuration specified by that reference is used.
```
Comment (dmuelle) [inline dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties:46]: what is using the ssl ref? Is it-

The ssl configuration that is specified by this reference is used in the gRPC client configuration.

File: dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties (hunk 6 above)
Comment (wtlucy) [reply dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties:31]: This is correct and sounds better, I'll update

File: dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties (hunk 7 above)
Comment (wtlucy) [reply dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties]: Yeah, this is correct.

File: dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+target.desc=The remote gRPC target URI, with wildcard support.
+
+headersToPropagate=HTTP header names to propagate
+headersToPropagate.desc=HTTP header names to propagate from an inbound request to an outbound call.
+
+clientInterceptors=GRPC client Interceptors
+clientInterceptors.desc=A list of fully qualified class names for gRPC client interceptor classes.
+
+authnToken=Authorization token
+authnToken.desc=The type of authorization token to propagate. Currently supported token types are jwt or oauth
```
Comment (dmuelle) [inline dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties:31]: Currently supported token types are jwt or oauth.

Needs a period- apologies if I left that off my suggestion

Comment (wtlucy) [reply dev/com.ibm.ws.grpc.client/resources/OSGI-INF/l10n/metatype.properties]: Good catch!  I deleted the period I added before - it's back now.

**Comment by wtlucy** [comment]
Thanks @dmuelle #build

**Comment by wtlucy** [comment]
#build

**Comment by wtlucy** [comment]
#build

**Comment by wtlucy** [comment]
#run-libby-bot

**Pull request merged**